web/models/workflow.ex

```diff
@@ -27,8 +28,10 @@ defmodule Designator.Workflow do
     query = from sms in "set_member_subjects",
       left_join: swc in "subject_workflow_counts", on: (sms.subject_id == swc.subject_id and swc.workflow_id == ^workflow_id),
       where: sms.subject_set_id == ^subject_set_id and is_nil(swc.retired_at),
-      select: sms.subject_id
+      select: {sms.subject_id, sms.priority}
```

Comment:
You're right, I should add test cases for null case. From what I can tell (correct me if i'm wrong), if `priority` is null `keysort` will spit out the results of the query as it was given from the db (assuming all the `set_member_subjects` from the query have `priority` null. If db returns  `set_member_subjects` where some have `priority` and some are null,  then it will first sort those that do have priority then append null priority `set_member_subjects` in the order given by db.

And you are right that this might affect randomness as it was working before, which isn't my intention and missed here. The more it's pointed out, the more I'm thinking I should move the sort by priority to `Sequential` module. 

Thank you for catching! @camallen

Reply:
> if priority is null keysort will spit out the results of the query as it was given from the db 

I assume so from the elixir docs (the sort is stable) https://hexdocs.pm/elixir/1.1.1/List.html#keysort/2

> (assuming all the set_member_subjects from the query have priority null.

In theory we should have priority for either all the subjects or none of the subjects but that's an ideal world :) It's currently not so easy to get the priority values consistently uploaded by researchers so we can end up with mixes of values being set on or missing.

> then it will first sort those that do have priority then append null priority set_member_subjects in the order given by db.

Yep - that sounds right. In theory for most of our random workflows this won't make a difference if the sort is stable and preserves the ordering from the DB (same behaviour as before) which is good to know but we should test it.

> The more it's pointed out, the more I'm thinking I should move the sort by priority to Sequential module.

TBH - if we can test/verify the ordering preservation on data loading for both cases (workflows for prioirty are set or missing) then we can most likely use this one implementation which seems simpler to me. 

I quite like this implementation as long as it doesn't change the overall system behaviour as it is simpler, load data and apply the needed transform for the downstream selection strategy (random vs sequential).

Reply:
> TBH - if we can test/verify the ordering preservation on data loading for both cases (workflows for priority are set or missing) then we can most likely use this one implementation which seems simpler to me.

Just to be clear, this would have to be a test for a staging environment, correct? I've added tests for null state cases on workflow model file test, but I think this is ready for full review. Let me know if I'm missing anything. I definitely will need to test this super rigorously to make sure these changes don't break what's currently working.

Reply:
there are a few ways we could do this - i'll reach out directly to discuss.